OrbitGl/ScopedStatus.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_SCOPED_STATUS_H_
+#define ORBIT_GL_SCOPED_STATUS_H_
+
+#include <string>
+#include <thread>
+
+#include "MainThreadExecutor.h"
+#include "OrbitBase/Logging.h"
+#include "StatusListener.h"
+
+/**
+ * This class holds scope for status. It also takes care
+ * of updating status on the main thread even if Update
+ * is called from a different thread.
+ *
+ * Usage example:
+ *
+ * {
+ *   ScopedStatus scoped_status =
+ *       new ScopedStatus(main_thread_executor, status_listener_, "Waiting for fish");
+ *
+ *   ...
+ *
+ *   scoped_status.Update("Found fish!");
+ *
+ *   ...
+ *
+ *   // Once out of scope it will clear the message.
+ * }
+ *
+ * See also StatusListener class.
+ */
+class ScopedStatus final {
+  // Instances fo this class are going to be moved a lot so we want data to be
```

Comment:
Nit: Add `private:` visibility explicitly - on initial skimming, I thought this was in the public section

Reply:
moved to private: section.